README.md

```diff
@@ -408,6 +408,9 @@ Check the [constraint rule comparison matrix](rule_comparison.md) for language-s
   
   // x must conform to a well known regex for HTTP header values (via RFC 7230) 
   string x = 1 [(validate.rules).string.well_known_regex = HTTP_HEADER_VALUE];
+  
+// x must conform to a well known regex that guards against \r\n\0 (via Envoy's validation)
```

Comment:
Would it make more sense to use a pattern similar  to validate rules string that acceps one of several options: (see above in this same file: `{suffix: ".proto", max_len:64}`) where he have a mode disallowing `\r\n\0`? I mean it still is a "header" value, just stricter.

Reply:
I love this suggestion! Not sure how much it would be used beyond `HTTP_HEADER_NAME`/`HTTP_HEADER_VALUE`, but i could add a `bool strict` mode for the regexes. I'll try it out.

Reply:
How does the usage of that look? `strict` should only affect these two regexes. Slightly displease that I can't upgrade enum KnownRegex to a message so I can put the qualifier there.

Ahhh I might be misinterpreting the meaning of strict/non-strict, but I think the RFC compliant ones are "strict" and `[^\r\n\0]` is "non-strict" since RFC-compliance disallows more. Or am I not not confusing negation :P

Reply:
I like the look of having well_known_regex, and a strict flag (even if it's only headers for now).

I think the negations here are incredibly confusing as well, but I think the way it is now is fine. Where `strict` disallows `^\r\n\0`, and where non strict allows them.

Reply:
(not sure if this is a misunderstanding but RFC compliant `HTTP_HEADER_NAME/HTTP_HEADER_VALUE` disallow most control characters (including \r\n\0), non-strict is just these control characters). but sounds good!